share/genbuild.sh

```diff
@@ -37,6 +37,8 @@ if [ "${BITCOIN_GENBUILD_NO_GIT}" != "1" ] && [ -e "$(command -v git)" ] && [ "$
     # otherwise generate suffix from git, i.e. string like "59887e8-dirty"
     SUFFIX=$(git rev-parse --short HEAD)
     git diff-index --quiet HEAD -- || SUFFIX="$SUFFIX-dirty"
+elif [ -f "$FILE" ]; then
```

Comment:
we already have a `if [ -f "$FILE" ]; then` check above, would it make sense to merge this into that?

Reply:
This condition handles a special case for gitian builds: with this PR the `obj/build.h` file is a part of the source tree that is fed to the `make`, and we do not want `genbuild.sh` to re-write it with the `// No build information available` string.

Reply:
How does it get updated for local build then, if the file already exists?

Reply:
> How does it get updated for local build then, if the file already exists?

For local build the `elif` condition is not applied as the previous `if` condition is evaluated to true.

Reply:
Ok, ths logic is kind of tricky to me. Even what it's supposed to do in what case. Maybe we need a test to make this clearer.